lib/gmp/gmp.nit

```diff
@@ -0,0 +1,370 @@
+        if l isa Long then
+            return (self <=> l) == 0
+        else
+            return false
+        end
+    end
+
+    redef fun <=(l) do return (self <=> l) <= 0
+    redef fun <(l) do return (self <=> l) < 0
+    redef fun >=(l) do return (self <=> l) >= 0
+    redef fun >(l) do return (self <=> l) > 0
+
+
+    #     assert(2.to_l + 2.to_l == 4.to_l)
+    redef fun +(l) do
+        var res = new NativeMPZ
+        val.add(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert(-(2.to_l) == (-2).to_l)
+    redef fun - do
+        var res = new NativeMPZ
+        val.neg res
+        return new Long(res)
+    end
+
+    #     assert(2.to_l - 2.to_l == 0.to_l)
+    redef fun -(l) do
+        var res = new NativeMPZ
+        val.sub(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert(2.to_l * 2.to_l == 4.to_l)
+    redef fun *(l) do
+        var res = new NativeMPZ
+        val.mul(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert(3.to_l / 2.to_l == 1.to_l)
+    redef fun /(l) do
+        var res = new NativeMPZ
+        val.tdiv_q(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert(5.to_l % 2.to_l == 1.to_l)
+    fun %(l: Long): Long do
+        var res = new NativeMPZ
+        val.mod(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert(3.to_l ^ 2 == 9.to_l)
+    fun ^(i: Int): Long do
+        var res = new NativeMPZ
+        var pow = new UInt64
+        pow.set_si i
+        val.pow_ui(res, pow)
+        pow.free
+        return new Long(res)
+    end
+
+    #     assert((-3).to_l.abs == 3.to_l)
+    fun abs: Long do
+        var res = new NativeMPZ
+        val.abs res
+        return new Long(res)
+    end
+
+    #     assert(15.to_l.gcd(10.to_l) == 5.to_l)
+    fun gcd(l: Long): Long do
+        var res = new NativeMPZ
+        val.gcd(res, l.val)
+        return new Long(res)
+    end
+
+    #     assert((0x10001).to_l.probab_prime(15) == 2)
+    fun probab_prime(reps: Int): Int do
+        return val.probab_prime_p(reps.to_i32)
+    end
+
+    #     assert(11.to_l.next_prime == 13.to_l)
+    fun next_prime: Long do
+        var res = new NativeMPZ
+        val.nextprime res
+        return new Long(res)
+    end
+
+    #     assert(11.to_l.zero == 0.to_l)
+    redef fun zero do return new Long(new NativeMPZ)
+
+    #     assert(11.to_l.value_of(4) == 4.to_l)
+    redef fun value_of(l) do return l.to_l
+
+    #     assert(11.to_l.to_i == 11)
+    redef fun to_i do return val.get_si
+
+    #     assert(11.to_l.to_f == 11.0)
+    redef fun to_f do return val.get_d
+
+    #     assert(11.to_l.to_s == "11")
+    redef fun to_s do
+        var cstr = val.get_str(10.to_i32)
+        var str = cstr.to_s
+        cstr.free
+        return str
+    end
+
+    #     assert(3.to_l.distance(6.to_l) == -3)
+    redef fun distance(l) do return (self - l).to_i
+
+    #     assert 3.to_l.is_between(2.to_l, 4.to_l)
+    redef fun is_between(c, d) do
```

Comment:
You don't need `is_between, min, or max` as they are already defined in lib/core/kernel.nit with pretty much the same code. (Same for `Ratio`)